### ISSUE TITLE
feat: add weight initialization method as a new design dimension

### ISSUE THREAD

**Issue by xiaohan2012** [xiaohan2012/stgym#106]
## Summary

Weight initialization is a well-known factor affecting GNN training dynamics (convergence speed, final accuracy, gradient flow). It is not currently exposed as a configurable parameter in STGym — all layers use PyTorch's default initialization. This issue tracks adding it as a proper design dimension so it can be included in the sweep.

## Current state

`GraphClassifierModelConfig` / `NodeClassifierModelConfig` in `stgym/config_schema.py` expose:
- `act`, `use_batchnorm`, `dropout`, `dim_inner`, `num_mp_layers`, etc.

There is no `weight_init` field. PyTorch defaults to **Kaiming uniform** for `Linear` and `Conv` layers.

## Proposed design choices

```yaml
design_dimension: model.weight_init
design_choices: [kaiming_uniform, kaiming_normal, xavier_uniform, xavier_normal, orthogonal]
```

Rationale for each:
- `kaiming_uniform` — PyTorch default; baseline
- `kaiming_normal` — common alternative for ReLU-family activations (pairs with `prelu`, `relu`, `swish`)
- `xavier_uniform` / `xavier_normal` — better suited for symmetric activations; worth testing with `prelu`/`swish`
- `orthogonal` — can improve gradient flow in deep networks; less common in GNNs but worth including

## Implementation outline

1. **`stgym/config_schema.py`**: add `WeightInitType = Literal[...]` and `weight_init: WeightInitType = "kaiming_uniform"` to the shared model config base.

2. **`stgym/model.py`**: add a `_apply_weight_init(module, method)` helper using `torch.nn.init.*` and call it via `self.apply(...)` after layer construction.

3. **`conf/exp/weight_init.yaml`**: new experiment config file following the existing pattern (see `conf/exp/activation.yaml`).

4. **`scripts/exp-weight_init/graph-clf.sh`** (and `node-clf.sh`): launch scripts following the existing `scripts/exp-{dim}/{task}.sh` convention.

## Notes

- Interaction with `use_batchnorm`: batch norm partly mitigates bad initialization by re-normalising activations — the effect of init may be more visible when `use_batchnorm=False`. Consider cross-referencing results with the `bn` dimension sweep.
- Interaction with `act`: Kaiming init assumes ReLU-like activations; Xavier is better for tanh/sigmoid. Worth noting in analysis.
- Pooling layers (DMoN, MinCut) have their own parameters — the init should apply to all `nn.Linear` layers uniformly for simplicity.

**Comment by xiaohan2012** [comment]
## Triage

**Type:** enhancement | **Priority:** P2 | **Status:** status/ready
**Blockers:** none
**Assessment:** Add weight initialization strategy as a new design dimension. Provides a principled lever for training-dynamics sweeps; implementation is localized to model config + layer construction.
